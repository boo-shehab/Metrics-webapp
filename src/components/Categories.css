
form input {
  width: 250px;
  font-size: 24px;
  border-radius: 50px;
  padding: 5px 10px;
}

.card-container {
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(250px,1fr));
  gap: 10px;
}
.card {
  border: 1px solid #474F7E;
  background-color: #5963C4;
  color: white;
  overflow: hidden;
  padding: 20px;
}