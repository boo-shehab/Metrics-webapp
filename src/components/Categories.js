import React, { useState } from 'react';
import CITY from '../data/city.json';
import City from './City';
import logo from '../image/logo.png';
import './Categories.css';

const Categories = () => {
  const [cities, setCities] = useState(CITY.slice(0, 12));

  const filtered = (ele) => {
    const filter = [];
    for (let i = 0; i < CITY.length; i += 1) {
      if (CITY[i].city.toLowerCase().includes(ele.target.value.toLowerCase())
      || CITY[i].country.toLowerCase().includes(ele.target.value.toLowerCase())) {
        filter.push(CITY[i]);
      }
      if (filter.length === 12) break;
    }
    setCities(filter);
  };

  return (
    <div className="container">
      <header>
        <img src={logo} alt="logo" className="logo" />
        <form>
          <input type="search" onChange={filtered} />
        </form>
      </header>
      <div>
        <div className="card-container">
          {cities.map((city, i) => <City key={city.id} city={city} index={i} />)}
        </div>
      </div>
    </div>
  );
};

export default Categories;
