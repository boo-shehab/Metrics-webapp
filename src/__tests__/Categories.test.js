import configureStore from 'redux-mock-store';
import React from 'react';
import { Provider } from 'react-redux';
import ReactTestRenderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Categories from '../components/Categories';

const mockStore = configureStore([]);

describe('renders Categories React-Redux Component', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      AirAPI: {
        loading: false,
        error: null,
        AirPollutions: [{
          coord: {
            lon: -0.1275,
            lat: 51.5072,
          },
          list: [
            {
              main: {
                aqi: 1,
              },
              components: {
                co: 226.97,
                no: 0,
                no2: 12.51,
                o3: 56.51,
                so2: 4.59,
                pm2_5: 5.22,
                pm10: 10.98,
                nh3: 0.34,
              },
              dt: 1672858800,
            },
          ],
        }],
      },
    });
    component = ReactTestRenderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Categories />
        </BrowserRouter>
      </Provider>,
    );
  });

  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
