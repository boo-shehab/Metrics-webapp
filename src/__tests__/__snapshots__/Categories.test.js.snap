// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Categories React-Redux Component should render with given state from Redux store 1`] = `
<div
  className="container"
>
  <header>
    <img
      alt="logo"
      className="logo"
      src="logo.png"
    />
    <form>
      <input
        onChange={[Function]}
        type="search"
      />
    </form>
  </header>
  <div>
    <div
      className="card-container"
    >
      <a
        href="/Details"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <div
          className="card"
        >
          <div>
            <div>
              city:
               
              London
            </div>
            <div>
              country:
               
              United Kingdom
            </div>
          </div>
          <div>
            date:
             
            04/01/2023
          </div>
          <div>
            air qualty:
             
            Good
          </div>
        </div>
      </a>
      <a
        href="/Details"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <div
          className="card"
        >
          <div>
            <div>
              city:
               
              Birmingham
            </div>
            <div>
              country:
               
              United Kingdom
            </div>
          </div>
          <div>
            date:
             
          </div>
          <div>
            air qualty:
             
          </div>
        </div>
      </a>
      <a
        href="/Details"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <div
          className="card"
        >
          <div>
            <div>
              city:
               
              Manchester
            </div>
            <div>
              country:
               
              United Kingdom
            </div>
          </div>
          <div>
            date:
             
          </div>
          <div>
            air qualty:
             
          </div>
        </div>
      </a>
      <a
        href="/Details"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <div
          className="card"
        >
          <div>
            <div>
              city:
               
              Leeds
            </div>
            <div>
              country:
               
              United Kingdom
            </div>
          </div>
          <div>
            date:
             
          </div>
          <div>
            air qualty:
             
          </div>
        </div>
      </a>
      <a
        href="/Details"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <div
          className="card"
        >
          <div>
            <div>
              city:
               
              Newcastle
            </div>
            <div>
              country:
               
              United Kingdom
            </div>
          </div>
          <div>
            date:
             
          </div>
          <div>
            air qualty:
             
          </div>
        </div>
      </a>
      <a
        href="/Details"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <div
          className="card"
        >
          <div>
            <div>
              city:
               
              Birstall
            </div>
            <div>
              country:
               
              United Kingdom
            </div>
          </div>
          <div>
            date:
             
          </div>
          <div>
            air qualty:
             
          </div>
        </div>
      </a>
      <a
        href="/Details"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <div
          className="card"
        >
          <div>
            <div>
              city:
               
              Glasgow
            </div>
            <div>
              country:
               
              United Kingdom
            </div>
          </div>
          <div>
            date:
             
          </div>
          <div>
            air qualty:
             
          </div>
        </div>
      </a>
      <a
        href="/Details"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <div
          className="card"
        >
          <div>
            <div>
              city:
               
              Liverpool
            </div>
            <div>
              country:
               
              United Kingdom
            </div>
          </div>
          <div>
            date:
             
          </div>
          <div>
            air qualty:
             
          </div>
        </div>
      </a>
      <a
        href="/Details"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <div
          className="card"
        >
          <div>
            <div>
              city:
               
              Portsmouth
            </div>
            <div>
              country:
               
              United Kingdom
            </div>
          </div>
          <div>
            date:
             
          </div>
          <div>
            air qualty:
             
          </div>
        </div>
      </a>
      <a
        href="/Details"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <div
          className="card"
        >
          <div>
            <div>
              city:
               
              Southampton
            </div>
            <div>
              country:
               
              United Kingdom
            </div>
          </div>
          <div>
            date:
             
          </div>
          <div>
            air qualty:
             
          </div>
        </div>
      </a>
      <a
        href="/Details"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <div
          className="card"
        >
          <div>
            <div>
              city:
               
              Nottingham
            </div>
            <div>
              country:
               
              United Kingdom
            </div>
          </div>
          <div>
            date:
             
          </div>
          <div>
            air qualty:
             
          </div>
        </div>
      </a>
      <a
        href="/Details"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <div
          className="card"
        >
          <div>
            <div>
              city:
               
              Bristol
            </div>
            <div>
              country:
               
              United Kingdom
            </div>
          </div>
          <div>
            date:
             
          </div>
          <div>
            air qualty:
             
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;
